### DISCUSSION TITLE
Request to have %MM to use full month name in search template #10559

### DISCUSSION THREAD

**Discussion by sebzapata** [pymedusa/Medusa#11287] [Ideas & Feature requests]
When creating a search template, it would be great to have the option to search by full month name.
For example, a show I like to download has the torrents listed as 'TV Show 8 May 2022' and I am unable to search for them.

Thanks

**Comment by Mika3578** [comment]
I have a similar request to the one posted in this discussion, but I also need support for the day in two decimal places, i.e., %Y, %MM,%DD. Can you confirm if the custom template feature is working well for you? Are the requests appearing properly in the Medusa logs?

**Comment by Mika3578** [comment]
[Here](https://www.ibm.com/docs/en/cmofz/10.1.0?topic=SSQHWE_10.1.0/com.ibm.ondemand.mp.doc/arsa0257.htm) an exemple of dates formats that can be implemented 😉
<html>
<body>
<div class="tablenoborder" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: 1.625rem; font-family: &quot;IBM Plex Sans&quot;, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline; color: rgb(22, 22, 22); letter-spacing: 0.16px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="bx--data-table-container" style="box-sizing: border-box; padding: 0.125rem 0px 0px; border: 0px; margin-top: 0px; margin-right: 0px; margin-bottom: 1.5rem !important; margin-left: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: 1.625rem; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline; position: relative; overflow: auto !important;">

Format Specifier | Format | Example
-- | -- | --
%m/%d/%Y | mm/dd/yyyy | 02/21/2018
%m/%d/%y | mm/dd/yy | 02/21/18
%d/%m/%Y | dd/mm/yyyy | 21/02/2018
%d/%m/%y | dd/mm/yy | 21/02/18
%d-%m-%Y | dd-mm-yyyy | 21-02-2018
%d-%m-%y | dd-mm-yy | 21-02-18
%m-%d-%Y | mm-dd-yyyy | 02-21-2018
%m-%d-%y | mm-dd-yy | 02-21-18
%Y-%m-%d | yyyy-mm-dd | 2018-02-21
%f/%e/%Y | m/d/yyyy | 2/21/2018
%f/%e/%y | m/d/yy | 2/21/18
%e/%f/%Y | d/m/yyyy | 21/2/2018
%e/%f/%y | d/m/yy | 21/2/18
%f-%e-%Y | m-d-yyyy | 2-21-2018
%f-%e-%y | m-d-yy | 2-21-18
%e-%f-%Y | d-m-yyyy | 21-2-2018
%e-%f-%y | d-m-yy | 21-2-18
%b %e, %Y | Mth d, yyyy | Feb 21, 2018
%B %e, %Y | Month d, yyyy | February 21, 2018
%b %d, %Y | Mth d, yyyy | Feb 21, 2018
%B %d, %Y | Month d, yyyy | February 21, 2018
%Y-%m-%d %H:%M:%S | yyyy-mm-dd hh:mm:ss | 2018-02-21 12:00:00
%H:%M:%S | hh:mm:ss | 12:00:00
%Y-%m-%d %I:%M:%S %p | yyyy-mm-dd HH:MM:SS tt | 2018-10-29 10:02:48 AM
%Y-%m-%d %I:%M:%S %p | yyyy-mm-dd HH:MM:SS tt | 2018-10-29 07:30:20 PM

</div></div><div class="section" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-variant-alternates: inherit; font-weight: 400; font-stretch: inherit; line-height: 1.625rem; font-family: &quot;IBM Plex Sans&quot;, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline; color: rgb(22, 22, 22); letter-spacing: 0.16px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><p class="p" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px 0px 1rem; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; line-height: 1.5; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 1rem; vertical-align: baseline; letter-spacing: 0px;">You can create a customized format specifier in the text portion of the list if the format of the date or date/time that you require is not one of the standard formats that are listed. Use a format specifier to describe the format of the date or time that you require. A format specifier consists of a set of values, such as<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%m</strong>, and separators, such as the / character. Specify the order in which the day, month, year, hour, minutes, and seconds appear. Exclude the day, month, year, hour, minutes, or seconds, if that part of the date or time does not appear in the format. Add value separators where required. For example,<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%d %B %Y</strong><span> </span>represents dates that appear in the format of<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">21 February 2018</strong>. As another example,<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%b %e %Y</strong><span> </span>represents dates that appear in the format of<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">Feb 5 2018</strong>. Date strings are not case sensitive. For example, you can specify<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%B</strong><span> </span>for dates that contain strings like February, FEBRUARY, or FEBruary.</p><p class="p" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px 0px 1rem; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; line-height: 1.5; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 1rem; vertical-align: baseline; letter-spacing: 0px;">When creating a customized format specifier, there must be one or more separator characters between numbers in a date. For example,<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%b %e, %Y</strong><span> </span>represents a date such as<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">Feb 5, 2018</strong>. The separator characters between the day of the month (5) and the year (2018) are a comma and a space. Specifying<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%b%e%Y</strong><span> </span>for a date such as<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">Feb52018</strong><span> </span>is not valid. However, the name of the month or the three-character abbreviation for the name of the month can be used as the separator between the day of the month and year. For example,<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%e%b%Y</strong><span> </span>represents dates that appear in the format of<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">5Feb2018</strong>.</p><p class="p" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px 0px 1rem; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; line-height: 1.5; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 1rem; vertical-align: baseline; letter-spacing: 0px;">The<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%e</strong><span> </span>and<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%d</strong><span> </span>can be used for one or two digit days. When a format specifier contains<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%e</strong>, single digit days are displayed as a single digit. When using<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%d</strong>, single digit days are displayed with a leading zero. For example,<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%b %e, %Y</strong><span> </span>will display the date<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">Feb 5, 2018</strong><span> </span>with a single digit for the day of the month. The format specifier<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%b %d, %Y</strong><span> </span>will display the date<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">Feb 05, 2018</strong><span> </span>with a leading zero for the day of the month.</p><p class="p" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px 0px 1rem; font-style: inherit; font-variant: inherit; font-weight: 400; font-stretch: inherit; line-height: 1.5; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 1rem; vertical-align: baseline; letter-spacing: 0px;">The<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%f</strong><span> </span>and<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%m</strong><span> </span>can be used for one or two digit months. When a format specifier contains<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%f</strong>, single digit months are displayed as a single digit. When using<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%m</strong>, single digit months are displayed with a leading zero. For example,<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%f %e %Y</strong><span> </span>will display the date<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">2 5 2018</strong><span> </span>with a single digit for the month. The format specifier<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">%m %d %Y</strong><span> </span>will display the date<span> </span><strong class="ph b" style="box-sizing: border-box; padding: 0px; border: 0px; margin: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-optical-sizing: inherit; font-kerning: inherit; font-feature-settings: inherit; font-variation-settings: inherit; font-size: 16px; vertical-align: baseline;">02 05 2018</strong><span> </span>with a leading zero for the month.</p></div>
</body>
</html>

**Reply by sebzapata** [reply]
Hi,
Thank you for your reply.
I'm a little confused as to what you're suggesting I do. Where can I create a custom format specifier?
I would like to add `%DD` for 9th/3rd etc

Thanks
